### DISCUSSION TITLE
Mangum enables you to do this with FastAPI

### DISCUSSION THREAD

**Discussion by badge** [harvey251/pydantic-lambda-handler#90] [General]
Hey, I'm currently in the process of working a similar tool for non-HTTP event handling, and stumbled across this project. I thought you might like to know that https://mangum.io/ enables you to do what you're trying to do here with FastAPI already: you define a FastAPI app and its routes as you would normally, and then just add Mangum as a handler at the end:

```python
from fastapi import FastAPI
from mangum import Mangum

app = FastAPI()


@app.get("/")
def read_root():
    return {"Hello": "World"}


@app.get("/items/{item_id}")
def read_item(item_id: int, q: str = None):
    return {"item_id": item_id, "q": q}

handler = Mangum(app, lifespan="off")
```

**Comment by harvey251** [comment]
HI, 

That's useful to know. I had looked into a few others like Chandler and AWS power tools but they didn't really do the"Fast api style type hints".

There's a couple of distinctions I needed to make.  

1. I needed to have greater separation between between dev and ops, only wanted to output a config file.
2. I needed to generate an accurate open api spec from code.
3. I needed a better way to pull args/kwargs the from Event and LambdaContext based on type hints.
4. Does not need to do the routing like fast api as we have API
5. Needed to be smaller and performant for lambdas
